### ISSUE TITLE
Number of coordinates in coordinate file does not match topology

### ISSUE THREAD

**Issue by luancarvalhomartins** [luancarvalhomartins/PyAutoFEP#6]
In some systems, running `gmx grompp` to prepare a .tpr to `gmx genion` in `prepare_complex_system` and `prepare_water_system` fails with: 
```
[ERROR] Failed to run gmx grompp. Error code 1.
(.......)
[ERROR] Fatal error:
[ERROR] number of coordinates in coordinate file
[ERROR] (/tmp/XXXX/build_system_SSMMHH_DDMMYYYY/protein/systemsolvated_step4_SSMMHH_DDMMYYYY.gro, NNNNN) does not match topology
[ERROR] (/tmp/XXXX/protein/build_system_SSMMHH_DDMMYYYY/systemsolv_step4_SSMMHH_DDMMYYYY.
top, NNNN)
```

This seems to happen when editing of _systemsolv_step4_SSMMHH_DDMMYYYY.
top_ by `gmx solvate` is somehow delayed or buffer in the /tmp filesystem.

**Comment by luancarvalhomartins** [comment]
Note that `running prepare_dual_topology.py` with `--output_hidden_temp_dir=False` should circumvent this issue.

**Comment by luancarvalhomartins** [comment]
Commit 84cdcbf346fa2c99284d8bc45214ee559d563161 didn't solve this.
Options I can see:

- Create a dir in `$PWD` and run `gmx solvate` and `gmx genion` in this dir, which would requiring copy files from build_dir to this dir in `$PWD`
- Add a message in case of `gmx grompp` fails suggesting the user to rerun with `--output_hidden_temp_dir=False` and referencing this issue.

**Comment by luancarvalhomartins** [comment]
Fixed by 3f7e7ee 
A error message was added when `gmx grompp` fails suggesting the user to run with `--output_hidden_temp_dir=False`
